python/taichi/lang/ops.py

```diff
@@ -39,8 +39,13 @@ def wrapped(a):
 def binary(foo):
     import taichi as ti
 
-    imp_foo = lambda x, y: foo(Expr(x), Expr(y))
-    rev_foo = lambda x, y: foo(Expr(y), Expr(x))
+    @functools.wraps(foo)
+    def imp_foo(x, y):
+        return foo(Expr(x), Expr(y))
+
+    @functools.wraps(foo)
+    def rev_foo(x, y):
+        return foo(Expr(y), Expr(x))
```

Comment:
I like lambdas too, but it turns out I have to write them into named functions to preserve the `__name__` of `foo` :thinking:

Reply:
Thank for fixing this!